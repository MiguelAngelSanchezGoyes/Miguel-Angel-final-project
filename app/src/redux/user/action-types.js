export const userActionTypes = {
  load: '@user/load',
  add: '@user/add',
  login: '@user/login',
  logout: '@user/logout',
  // verify: '@user/verify',
};
