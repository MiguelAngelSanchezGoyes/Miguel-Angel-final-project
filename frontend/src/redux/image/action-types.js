export const imageActionTypes = {
  load: '@image/load',
  add: '@image/add',
  update: '@image/update',
  remove: '@image/remove',
};
