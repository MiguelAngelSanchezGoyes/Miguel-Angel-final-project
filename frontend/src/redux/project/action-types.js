export const projectActionTypes = {
  load: '@project/load',
  loadById: '@project/loadById',
  add: '@project/add',
  edit: '@project/edit',
  remove: '@project/remove',
};
