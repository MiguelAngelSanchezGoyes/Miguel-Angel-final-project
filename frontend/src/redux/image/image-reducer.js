/* eslint-disable no-underscore-dangle */
import { imageActionTypes } from './action-types';

const initialImage = {
  images: [],
  image: {},
};

const imageReducer = (state = initialImage, action) => {
  switch (action.type) {
    case imageActionTypes.load:
      return { ...state, images: [...action.payload] };
    case imageActionTypes.add:
      return { ...state, image: action.payload };
    case imageActionTypes.remove:
      return {
        ...state,
        projects: state.images.filter((item) => item._id !== action.id),
      };
    default:
      return state;
  }
};

export default imageReducer;
