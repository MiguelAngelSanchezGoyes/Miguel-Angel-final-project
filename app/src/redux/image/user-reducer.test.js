import { imageReducer } from './image-reducer';
import { imageActionTypes } from './action-types';

describe('userReducer reducer', () => {
  const initialState = [];
  test('should return the initial state', () => {
    expect(imageReducer(undefined, [])).toMatchSnapshot();
  });

  test('should handle imageReducer.add', () => {
    expect(
      imageReducer(initialState, {
        type: imageActionTypes.add,
      })
    ).toMatchSnapshot();
  });

  test('should handle imageReducer.load', () => {
    expect(
      imageReducer(initialState, {
        type: imageActionTypes.load,
      })
    ).toMatchSnapshot();
  });

  test('should handle imageReducer.update', () => {
    expect(
      imageReducer(initialState, {
        type: imageActionTypes.update,
      })
    ).toMatchSnapshot();
  });

  test('should handle imageReducer.remove', () => {
    expect(
      imageReducer(initialState, {
        type: imageActionTypes.remove,
      })
    ).toMatchSnapshot();
  });
});
