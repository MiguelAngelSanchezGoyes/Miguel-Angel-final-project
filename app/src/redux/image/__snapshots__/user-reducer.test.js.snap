// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer reducer should handle imageReducer.add 1`] = `
Object {
  "image": undefined,
}
`;

exports[`userReducer reducer should handle imageReducer.update 1`] = `Array []`;

exports[`userReducer reducer should return the initial state 1`] = `
Object {
  "image": Object {},
  "images": Array [],
}
`;
